lib/pure/math.nim

```diff
@@ -880,6 +880,33 @@ func floorMod*[T: SomeNumber](x, y: T): T =
   result = x mod y
   if (result > 0 and y < 0) or (result < 0 and y > 0): result += y
 
+func euclDiv*[T: SomeInteger](x, y: T): T {.since: (1, 5, 1).} =
+  ## Returns euclidean division of `x` by `y`.
+  runnableExamples:
+    assert euclDiv(13, 3) == 4
+    assert euclDiv(-13, 3) == -5
+    assert euclDiv(13, -3) == -4
+    assert euclDiv(-13, -3) == 5
+  result = x div y
+  let r = x mod y
+  if r < 0:
```

Comment:
```suggestion
  if x mod y < 0:
```

Reply:
I need to test these two programs to see whether they are the same.

Reply:
It seems the same.